### ISSUE TITLE
Is there a way we can explicit check if we shouldn't call getLengthSync?

### ISSUE THREAD

**Issue by bitinn** [form-data/form-data#196]
We use `getLengthSync` to add `content-length` automatically, but in the case where you have streams, you can't really call `getLengthSync`, you need to use `getLength` callback to figure out the `content-length`.

While I do like `form-data` to explicitly encourage using `getLength` in readme, having a way so that we don't call `getLengthSync` when it's not suitable would be great too.

Not that we can't fallback to `getLength`, but we rather not make this hack to support `form-data` too complex.

ref:
- https://github.com/bitinn/node-fetch/issues/102
- https://github.com/bitinn/node-fetch/blob/master/index.js#L94-L96

**Comment by bitinn** [comment]
In addition to this, it would be great if we can actually detect the body is from `form-data` instead of guessing it by checking method names...

https://github.com/bitinn/node-fetch/blob/master/index.js#L85-L88

**Comment by bitinn** [comment]
At the moment `this._lengthRetrievers.length` seems like the best indicator, but since it's intended to be private, it would be great if we can get a public getter for it.

Alternatively, when user has set `knownLength`, should we allow `getLengthSync` to return the `content-length` instead of throwing errors, since we already know the length at this point?

**Comment by alexindigo** [comment]
@bitinn Thanks for the ideas, do you mind to provide some code snippet to illustrate your vision for FormData? 

And one thing to keep in mind, FormData couldn't have knowledge about all types of streams, it's already doing too much. So the best way would be to expose convenience methods to allow such things in "userland".

Thanks.

**Comment by bitinn** [comment]
```
FormData.prototype.hasAsyncLength = function() {
  return this._lengthRetrievers.length > 0
}
```

I am bad at naming stuffs. But this should do. Basically this is what triggered the error on getLengthSync.

**Comment by bitinn** [comment]
It's been a while, do you suggest we go for `this._lengthRetrievers.length` instead? I see no other way to verify if a form-data object support `getLengthSync` or not.

**Comment by alexindigo** [comment]
Fixed in v2.1.0
